src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -323,9 +323,16 @@ public SourceLocation GetLocationByIl(int pos)
             SequencePoint prev = null;
             foreach (SequencePoint sp in DebugInformation.SequencePoints)
             {
-                if (sp.Offset > pos)
+                if (sp.Offset > pos) {
+                    //get the earlier line number if the offset is in a hidden sequence point and has a earlier line number available
+                    // if is doesn't continue and get the next line number that is not in a hidden sequence point
+                    if (sp.IsHidden && prev == null)
+                        continue;
                     break;
-                prev = sp;
+                }
+
+                if (!sp.IsHidden)
+                    prev = sp;
```

Comment:
nit: if you moved this to the beginning of the foreach you could make the condition of if (prev == null) instead of checking sp.IsHidden twice.

Reply:
IIUC, I cannot do this, otherwise it will replace the `prev` even if `sp.Offset > pos`, and it shouldn't, we should break before it.